File: lab-1/GraphAlgo/CSharp/EeeeGraphAlgo/EeeeGraphAlgo/Node.cs
namespace EeeeGraphAlgo
{
    public class Node
    {
        public Node(int id)
        {
            Id = id;
            IsVisited = false;
            Children = new List<Node>();
        }

        public int Id { get; }
        
        public bool IsVisited { get; set; }
        
        public List<Node> Children { get; }
    }   
}

File: lab-1/GraphAlgo/Java/EeeeGraphAlgo/src/main/java/eeee/dgar/Node.java
package eeee.dgar;

import java.util.ArrayList;
import java.util.List;

public class Node
{
    private int _id;
    private boolean _isVisited;
    private List<Node> _children;

    public Node(int id)
    {
        _id = id;
        _isVisited = false;
        _children = new ArrayList<>();
    }

    public int getId()
    {
        return _id;
    }

    public boolean isVisited()
    {
        return _isVisited;
    }

    public void setIsVisited(boolean isVisited)
    {
        _isVisited = isVisited;
    }

    public List<Node> children()
    {
        return _children;
    }
}